### PR TITLE
Bugfix for track selector

### PR DESCRIPTION
**Pull request by dkobylianskii**
## Summary

This pull request introduces the following changes

* Set unsigned variables to 0 in order to handle cases like `numberOfSCTSharedHits`
* Update mock tracks variable dtypes to be consistent with TDD output

## Conformity
- [x] [Changelog entry](https://github.com/umami-hep/atlas-ftag-tools/blob/main/changelog.md)
- [x] [Documentation](https://umami-hep.github.io/atlas-ftag-tools/)

### CODE REVIEW

File: ftag/track_selector.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def __call__(self, tracks: np.ndarray) -> np.ndarray:
             if issubclass(tracks[var].dtype.type, np.floating):
                 tracks[var][rm_idx] = np.nan
             elif issubclass(tracks[var].dtype.type, np.integer):
-                tracks[var][rm_idx] = -1
+                tracks[var][rm_idx] = 0
```
Comment (samvanstroud) [inline ftag/track_selector.py:45]: What's the motivation for using 0 over -1? -1 should look more like the padded tracks no?

Comment (dkobylianskii) [reply ftag/track_selector.py:45]: -1 doesn't work with unsigned types from H5 files.
For example `numberOfSCTSharedHits` has `unsigned char` type

Comment (samvanstroud) [reply ftag/track_selector.py:45]: ah yeah good thinking :) ok and for signed either should be fine as we set the valid flag anyways. Ok I'll merge this!

**Comment by samvanstroud** [comment]
Great thanks @dkobylianskii!

**Pull request merged**